web/src/narrow.js

```diff
@@ -87,8 +87,6 @@ export function compute_narrow_title(filter) {
         if (inbox_util.is_visible()) {
             return $t({defaultMessage: "Inbox"});
         }
-
-        return $t({defaultMessage: "All messages"});
```

Comment:
Added a bonus commit to remove this as it's now dead code following the previous commit; also cleaned up some tests. 

It looks like we have 4 other tests senselessly mocking `recent_view_util` when they could just call `set_visible`; @amanagr would you be up for doing a follow-up sweep to remove those?

Reply:
Opened #27082